flink-connector/src/main/java/com/google/pubsub/flink/PubSubSink.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.pubsub.flink;
+
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.auth.Credentials;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.pubsub.flink.internal.sink.PubSubFlushablePublisher;
+import com.google.pubsub.flink.internal.sink.PubSubPublisherCache;
+import com.google.pubsub.flink.internal.sink.PubSubSinkWriter;
+import com.google.pubsub.v1.TopicName;
+import java.io.IOException;
+import java.util.Optional;
+import org.apache.flink.api.connector.sink2.Sink;
+import org.apache.flink.api.connector.sink2.SinkWriter;
+import org.apache.flink.util.Preconditions;
+
+public class PubSubSink<T> implements Sink<T> {
```

Comment:
just use AutoValue for this to enforce the constraints.

Reply:
Done